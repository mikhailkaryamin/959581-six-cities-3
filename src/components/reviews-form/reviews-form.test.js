import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, {
  mount
} from "enzyme";
import ReviewsForm from "./reviews-form.jsx";

Enzyme.configure({
  adapter: new Adapter(),
});

const comment = `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Curabitur ornare nunc a blandit ultricies.
  Integer a urna in nunc congue efficitur.`;
const onChange = () => {};
const onSubmit = () => {};
const rating = 3;

test(`Render reviews form`, () => {
  const tree = mount(
      <ReviewsForm
        comment={comment}
        onChange={onChange}
        onSubmit={onSubmit}
        rating={rating}
      />
  ).html();

  expect(tree).toMatchSnapshot();
});
