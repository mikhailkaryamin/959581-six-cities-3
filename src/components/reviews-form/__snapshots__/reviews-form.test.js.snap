// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render reviews form 1`] = `
"<form class=\\"reviews__form form\\" action=\\"#\\" method=\\"post\\"><label class=\\"reviews__label form__label\\" for=\\"review\\">Your review</label><div class=\\"reviews__rating-form form__rating\\"><input class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" id=\\"5-stars\\" type=\\"radio\\" value=\\"5\\"><label for=\\"5-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" id=\\"4-stars\\" type=\\"radio\\" value=\\"4\\"><label for=\\"4-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" id=\\"3-stars\\" type=\\"radio\\" value=\\"3\\" checked=\\"\\"><label for=\\"3-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" id=\\"2-stars\\" type=\\"radio\\" value=\\"2\\"><label for=\\"2-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" id=\\"1-stars\\" type=\\"radio\\" value=\\"1\\"><label for=\\"1-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label></div><textarea class=\\"reviews__textarea form__textarea\\" id=\\"review\\" name=\\"review\\" placeholder=\\"Tell how was your stay, what you like and what can be improved\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Curabitur ornare nunc a blandit ultricies.
  Integer a urna in nunc congue efficitur.</textarea><div class=\\"reviews__button-wrapper\\"><p class=\\"reviews__help\\">To submit review please make sure to set<span class=\\"reviews__star\\">rating</span>and describe your stay with at least<b class=\\"reviews__text-amount\\">50 characters</b>.</p><button class=\\"reviews__submit form__submit button\\" type=\\"submit\\" disabled=\\"\\">Submit</button></div></form>"
`;
